### PR TITLE
refactor(modularize): Add `hosts/common.nix` for common nixos settings

### PR DESCRIPTION
**Pull request by AVGVSTVS96**
### TL;DR

Refactored NixOS configuration by extracting common settings into a shared file.

### What changed?

- Created a new `nixos-common.nix` file that contains shared configuration for all NixOS systems
- Moved common settings from individual host configurations to this shared file:
  - User configuration
  - Boot loader settings
  - Time zone
  - Hardware configuration
  - Sudo rules
  - Font packages
  - Common system packages
- Updated both `nixos/default.nix` and `vm/default.nix` to import the common configuration
- Removed duplicated code from host-specific files
- Added comments for clarity on future improvements to networking configuration

### CODE REVIEW

**Comment by AVGVSTVS96** [comment]
* **#13** <a href="https://app.graphite.dev/github/pr/AVGVSTVS96/nixos-config/13?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#12** <a href="https://app.graphite.dev/github/pr/AVGVSTVS96/nixos-config/12?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/AVGVSTVS96/nixos-config/12?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#11** <a href="https://app.graphite.dev/github/pr/AVGVSTVS96/nixos-config/11?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#9** <a href="https://app.graphite.dev/github/pr/AVGVSTVS96/nixos-config/9?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#10](https://github.com/AVGVSTVS96/nixos-config/pull/10) <a href="https://app.graphite.dev/github/pr/AVGVSTVS96/nixos-config/10?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#7** <a href="https://app.graphite.dev/github/pr/AVGVSTVS96/nixos-config/7?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5** <a href="https://app.graphite.dev/github/pr/AVGVSTVS96/nixos-config/5?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#6](https://github.com/AVGVSTVS96/nixos-config/pull/6) <a href="https://app.graphite.dev/github/pr/AVGVSTVS96/nixos-config/6?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#4** <a href="https://app.graphite.dev/github/pr/AVGVSTVS96/nixos-config/4?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2** <a href="https://app.graphite.dev/github/pr/AVGVSTVS96/nixos-config/2?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#3](https://github.com/AVGVSTVS96/nixos-config/pull/3) <a href="https://app.graphite.dev/github/pr/AVGVSTVS96/nixos-config/3?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Comment by AVGVSTVS96** [comment]
### Merge activity

* **Apr 21, 5:56 PM EDT**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/AVGVSTVS96/nixos-config/12).
* **Apr 21, 5:57 PM EDT**: A user merged this pull request with [Graphite](https://app.graphite.dev/github/pr/AVGVSTVS96/nixos-config/12).
